ostd/src/task/mod.rs

```diff
@@ -189,7 +189,7 @@ impl TaskOptions {
         extern "C" fn kernel_task_entry() {
             let current_task = current_task()
                 .expect("no current task, it should have current task in kernel task entry");
-            current_task.func.call(());
+            let _ = catch_unwind_injected(&current_task.func);
             current_task.exit();
```

Comment:
We should expose `catch_unwind_injected` to allow OSTD users to use it directly. This is particularly useful when a POSIX thread panics in the kernel space. In this case, it is insufficient to only exit the panicking POSIX thread; the entire process should be killed.